### PR TITLE
feat: add support for returning revoked members

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

Screenshot 2024-02-07 at 11 33 58

fixes https://github.com/dust-tt/tasks/issues/368

## Risk

Blast radius: onboarding, signup, login (inverted 80/20)

### CODE REVIEW

**Review by philipperolet — APPROVED** [review]
LGTM :pray:

File: front/pages/api/create-new-workspace.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import type { WithAPIErrorReponse } from "@dust-tt/types";
```
Comment (flvndvd) [inline front/pages/api/create-new-workspace.ts:1]: Most of the files seems to be using snake case convention in this folder.

File: front/pages/api/create-new-workspace.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import { authOptions } from "./auth/[...nextauth]";
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorReponse<{ sId: string }>>
+): Promise<void> {
+  const session = await getServerSession(req, res, authOptions);
+  if (!session) {
+    res.status(401).end();
```
Comment (flvndvd) [inline front/pages/api/create-new-workspace.ts:18]: `nit`.
```suggestion
    res.sendStatus(401);
```

File: front/pages/no-workspace.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    where: { userId: user.id },
+  });
+
+  if (!memberships.length) {
+    const message = "Unreachable: user has no memberships";
+    logger.error({ userId: user.id, panic: true }, message);
+    throw new Error(message);
+  }
+
+  if (user.workspaces.length) {
```
Comment (flvndvd) [inline front/pages/no-workspace.tsx:39]: `nit`
```suggestion
  if (user.workspaces.length > 0) {
```

File: front/pages/no-workspace.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    return {
+      notFound: true,
+    };
+  }
+
+  const memberships = await Membership.findAll({
+    where: { userId: user.id },
+  });
+
+  if (!memberships.length) {
```
Comment (flvndvd) [inline front/pages/no-workspace.tsx:33]: `nit`
```suggestion
  if (memberships.length === 0) {
```

File: front/pages/no-workspace.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+  const onCreateWorkspace = async () => {
+    const res = await fetch("/api/create-new-workspace", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    if (!res.ok) {
+      console.error("Failed to create new workspace");
```
Comment (flvndvd) [inline front/pages/no-workspace.tsx:82]: Should we keep this log?

File: front/pages/no-workspace.tsx (hunk 4 above)
Comment (fontanierh) [reply front/pages/no-workspace.tsx:33]: Same questions

File: front/pages/no-workspace.tsx (hunk 5 above)
Comment (fontanierh) [reply front/pages/no-workspace.tsx:82]: Can replace with a sendNotification maybe

File: front/pages/api/create-new-workspace.ts (hunk 1 above)
Comment (fontanierh) [reply front/pages/api/create-new-workspace.ts:1]: We kind of have a mix in `/pages` between snake and kebab. IMHO kebab is the norm for endpoints

**Comment by fontanierh** [comment]
@flvndvd 

> Error: pages/api/create-new-workspace.ts(18,9): error TS2339: Property 'sendStatus' does not exist on type 'NextApiResponse<WithAPIErrorReponse<{ sId: string; }>>'.

**Pull request merged**